File: app/src/main/java/app/islammedia/halaqatalquran/Database/StudentHomeWorks.java
package app.islammedia.halaqatalquran.Database;

import androidx.room.Embedded;
import androidx.room.Junction;
import androidx.room.Relation;

import java.util.List;

class StudentHomeWorks{

    @Embedded public Student st;
    @Relation(
            parentColumn = "idName",
            entityColumn = "idTitle",
            associateBy = @Junction(StudentsAndHomeWorksCrossRef.class)
    )
    public List<HomeWork> hws;

}

File: app/src/main/java/app/islammedia/halaqatalquran/database/helpers/StudentHomeWorks.kt
package app.islammedia.halaqatalquran.database.helpers

import androidx.room.Embedded
import androidx.room.Junction
import androidx.room.Relation
import app.islammedia.halaqatalquran.database.entities.HomeWork
import app.islammedia.halaqatalquran.database.entities.Student
import app.islammedia.halaqatalquran.database.entities.StudentsAndHomeWorksCrossRef

class StudentHomeWorks {
    @kotlin.jvm.JvmField
    @Embedded
    var st: Student? = null

    @kotlin.jvm.JvmField
    @Relation(parentColumn = "idName", entityColumn = "idTitle", associateBy = Junction(StudentsAndHomeWorksCrossRef::class))
    var hws: MutableList<HomeWork?>? = null
}